doc/source/whatsnew/v2.2.2.rst

```diff
@@ -24,6 +24,7 @@ Bug fixes
 ~~~~~~~~~
 - :meth:`DataFrame.__dataframe__` was showing bytemask instead of bitmask for ``'string[pyarrow]'`` validity buffer (:issue:`57762`)
 - :meth:`DataFrame.__dataframe__` was showing non-null validity buffer (instead of ``None``) ``'string[pyarrow]'`` without missing values (:issue:`57761`)
+- :meth:`DataFrame.to_sql` was failing to find the right table on PostgreSQL when using the ``public`` schema (:issue:`57539`)
```

Comment:
I think this technically would have affected any database and any schema, so we don't need to be overly specific about it being Postgres with the public schema